packages/ui/src/views/repo/repo-commits/components/commits-list.tsx

```diff
@@ -1,72 +1,68 @@
-import { useMemo } from 'react'
+import { FC, useMemo } from 'react'
 
-import { Avatar, AvatarFallback, CommitCopyActions, NodeGroup, StackedList, Text } from '@/components'
+import { Avatar, AvatarFallback, CommitCopyActions, NodeGroup, StackedList } from '@/components'
 import { formatDate, getInitials } from '@/utils/utils'
+import { TypesCommit } from '@/views'
 
-import { TypesCommit } from '../types'
+type CommitsGroupedByDate = Record<string, TypesCommit[]>
 
 interface CommitProps {
   data?: TypesCommit[]
 }
 
-export const CommitsList = ({ ...props }: CommitProps) => {
-  const data = props.data
-  const commitsGroupedByDate: Record<string, TypesCommit[]> = useMemo(
-    () =>
-      data?.reduce(
-        (group, commit) => {
-          const date = formatDate(commit.committer?.when as string)
+export const CommitsList: FC<CommitProps> = ({ data }) => {
+  const entries = useMemo(() => {
+    const commitsGroupedByDate = !data
+      ? {}
+      : data.reduce<CommitsGroupedByDate>((group, commit) => {
+          const date = formatDate(commit.committer?.when ?? '')
           group[date] = (group[date] || []).concat(commit)
           return group
-        },
-        {} as Record<string, TypesCommit[]>
-      ) || {},
-    [data]
-  )
+        }, {})
+
+    return Object.entries(commitsGroupedByDate) as [string, TypesCommit[]][]
```

Comment:
Looks like you don't need the type override here, TS seems to infer it properly.
```suggestion
    return Object.entries(commitsGroupedByDate)
```

Reply:
Perhaps this is an issue specific to my IDE

![image](https://github.com/user-attachments/assets/c84a748f-d7f4-4f51-a7bc-ba37f463ab01)